AdyenActions/Components/QRCode/QRCodeActionComponent.swift

```diff
@@ -84,35 +84,42 @@ public final class PixActionComponent: ActionComponent, Cancellable {
     /// - Parameter timeoutInterval: QR Code expiration timeout
     internal init(context: AdyenContext,
                   configuration: Configuration = .init(),
-                  pollingComponentBuilder: AnyPollingHandlerProvider,
+                  pollingComponentBuilder: AnyPollingHandlerProvider? = nil,
                   timeoutInterval: TimeInterval) {
         self.context = context
         self.configuration = configuration
         self.pollingComponentBuilder = pollingComponentBuilder
         self.expirationTimeout = timeoutInterval
-        
+
         updateExpiration(timeoutInterval)
     }
 
     /// Handles QR code action.
     ///
     /// - Parameter action: The QR code action.
     public func handle(_ action: QRCodeAction) {
-        let pollingComponent = pollingComponentBuilder.handler(for: action.paymentMethodType)
-        pollingComponent.delegate = self
-        
-        assert(presentationDelegate != nil)
+        let pollingComponent = pollingComponentBuilder?.handler(for: action.paymentMethodType)
```

Comment:
`pollingComponent` is not retained, it will get deallocated immediately after the function returns, :-)

Reply:
This is a nice catch. But apparently it's been like this for a while. Has it not been working all this time? 🤔

Reply:
oh wow, shit, that's definitely a bug :-D

Reply:
fixed this bug. Thank you guys for identifying this. Teamwork makes the dream work 🕺